client/a8c-for-agencies/sections/sites/site-actions/use-site-actions.tsx

```diff
@@ -0,0 +1,467 @@
+				isEnabled: canDelete,
+			},
+		];
+	}, [
+		canDelete,
+		canRemove,
+		dispatch,
+		isDevSite,
+		isLargeScreen,
+		isWPCOMSimpleSite,
+		isWPCOMSite,
+		onSelect,
+		setDataViewsState,
+		setSelectedSiteFeature,
+		site?.value,
+		siteError,
+		siteValue,
+		translate,
+	] );
+}
+
+type SiteActions = {
+	isLargeScreen: boolean;
+	onRefetchSite?: () => Promise< unknown >;
+	setDataViewsState: ( callback: ( prevState: DataViewsState ) => DataViewsState ) => void;
+	setSelectedSiteFeature: ( siteFeature: string | undefined ) => void;
+};
+
+const createRemoveSiteActionModal =
+	( onRefetchSite: any ) =>
+	( {
+		items,
+		closeModal,
+		onActionPerformed,
+	}: {
+		items: SiteData[];
+		closeModal: () => void;
+		onActionPerformed?: () => void;
+	} ) => {
+		const translate = useTranslate();
+		const dispatch = useDispatch();
+		const { mutate: removeSite } = useRemoveSiteMutation();
+
+		const item = items[ 0 ];
+		const siteName = item.site.value.url;
+		const siteId = item.site.value.a4a_site_id;
+
+		const onRemoveSite = () => {
+			if ( ! siteId ) {
+				return;
+			}
+
+			removeSite(
+				{ siteId },
+				{
+					onSuccess: () => {
+						// Add 1 second delay to refetch sites to give time for site profile to be reindexed properly.
+						setTimeout( () => {
+							onRefetchSite?.()?.then( () => {
+								closeModal?.();
+								onActionPerformed?.();
+								dispatch( successNotice( translate( 'The site has been successfully removed.' ) ) );
+							} );
+						}, 1000 );
+					},
+				}
+			);
+		};
+
+		const onSubmit = ( event: React.FormEvent ) => {
+			event.preventDefault();
+			onRemoveSite();
+		};
+
+		return (
+			<form onSubmit={ onSubmit }>
+				{ translate(
+					'Are you sure you want to remove the site {{b}}%(siteName)s{{/b}} from the dashboard?',
+					{
+						args: { siteName },
+						components: {
+							b: <b />,
+						},
+						comment: '%(siteName)s is the site name',
+					}
+				) }
+				<HStack justify="right">
+					<Button
+						__next40pxDefaultSize
+						variant="tertiary"
+						onClick={ () => {
+							closeModal?.();
+						} }
+					>
+						{ translate( 'Cancel' ) }
+					</Button>
+					<Button __next40pxDefaultSize variant="primary" type="submit" isDestructive>
+						{ translate( 'Remove site' ) }
+					</Button>
+				</HStack>
+			</form>
+		);
+	};
+
+export function useSiteActionsDataViews( {
+	isLargeScreen,
+	onRefetchSite,
+	setDataViewsState,
+	setSelectedSiteFeature,
+}: SiteActions ) {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	return useMemo( () => {
+		const isUrlOnly = ( item: SiteData ) =>
+			item.site?.value?.sticker?.includes( 'jetpack-manage-url-only-site' );
+		const isWPComSite = ( item: SiteData ) =>
+			item.site.value.is_atomic || item.site.value.is_simple;
+		const isDevSite = ( item: SiteData ) => item.site.value.a4a_is_dev_site;
+		const hasRemoveManagedSitesCapability = () => {
+			// TODO: implement this based on the item data.
```

Comment:
This needs to be implemented.

Reply:
Done at https://github.com/Automattic/wp-calypso/pull/98276/commits/7c46dbf278a46da88cb4fd8a11768baab7341e49